examples/deepResearch/deepResearch.tsx

```diff
@@ -0,0 +1,117 @@
+import { ChatCompletion, OpenAIProvider } from "@gensx/openai";
+import { gsx } from "gensx";
+
+import { ArxivEntry, ArxivSearch } from "./arxiv.js";
+import { GradeDocuments } from "./grader.js";
+import { PromptToQuery } from "./promptToQuery.js";
+import { FetchAndSummarize, FetchAndSummarizeOutput } from "./summarize.js";
+
+interface FindResearchProps {
+  prompt: string;
+}
+
+export const FindResearch = gsx.Component<FindResearchProps, ArxivEntry[]>(
+  ({ prompt }) => (
+    <OpenAIProvider apiKey={process.env.OPENAI_API_KEY}>
```

Comment:
You don't need to have this inside the component, you only need it in one place in the component tree (probably at the outermost point).

Reply:
good catch! ended up removing this component completely